Not compatible with openssl 1.1.1?

Compiling on Ubuntu 20.04 with OpenSSL 1.1.1f fails:
```
tony@desk1:~/development/hexpand$ make test
clang -c -Wall -Werror -Wno-deprecated-declarations  hexpand.c -o hexpand.o
hexpand.c:22:27: error: incomplete definition of type 'struct evp_md_ctx_st'
        int length_modulo = mdctx->digest->block_size;
                            ~~~~~^
/usr/include/openssl/ossl_typ.h:92:16: note: forward declaration of 'struct evp_md_ctx_st'
typedef struct evp_md_ctx_st EVP_MD_CTX;
               ^
hexpand.c:32:63: error: incomplete definition of type 'struct evp_md_ctx_st'
        unsigned char* h_data = (unsigned char *)((SHA512_CTX *)mdctx->md_data)->h;
                                                                ~~~~~^
/usr/include/openssl/ossl_typ.h:92:16: note: forward declaration of 'struct evp_md_ctx_st'
typedef struct evp_md_ctx_st EVP_MD_CTX;
               ^
hexpand.c:33:26: error: incomplete definition of type 'struct evp_md_ctx_st'
        int h_data_size = (mdctx->digest->md_size);
                           ~~~~~^
/usr/include/openssl/ossl_typ.h:92:16: note: forward declaration of 'struct evp_md_ctx_st'
typedef struct evp_md_ctx_st EVP_MD_CTX;
               ^
hexpand.c:47:70: error: format specifies type 'unsigned int' but the argument has type '__uint64_t' (aka 'unsigned long') [-Werror,-Wformat]
        if (length_modulo == 128) sprintf(output+2*padding, "%032" PRIx32 , htole64(8*length));
                                                             ~~~~~~~~~~~~   ^~~~~~~~~~~~~~~~~
/usr/include/endian.h:50:22: note: expanded from macro 'htole64'
#  define htole64(x) __uint64_identity (x)
                     ^~~~~~~~~~~~~~~~~~~~~
hexpand.c:48:49: error: format specifies type 'unsigned int' but the argument has type '__uint64_t' (aka 'unsigned long') [-Werror,-Wformat]
        else sprintf(output+2*padding, "%016" PRIx32 , htole64(8*length));
                                        ~~~~~~~~~~~~   ^~~~~~~~~~~~~~~~~
/usr/include/endian.h:50:22: note: expanded from macro 'htole64'
#  define htole64(x) __uint64_identity (x)
                     ^~~~~~~~~~~~~~~~~~~~~
hexpand.c:55:27: error: incomplete definition of type 'struct evp_md_ctx_st'
        int length_modulo = mdctx->digest->block_size;
                            ~~~~~^
/usr/include/openssl/ossl_typ.h:92:16: note: forward declaration of 'struct evp_md_ctx_st'
typedef struct evp_md_ctx_st EVP_MD_CTX;
               ^
hexpand.c:64:19: error: incomplete definition of type 'struct evp_md_ctx_st'
        ((MD5_CTX *)mdctx->md_data)->A = htonl(strntoul(signature, 8, 16));
                    ~~~~~^
/usr/include/openssl/ossl_typ.h:92:16: note: forward declaration of 'struct evp_md_ctx_st'
typedef struct evp_md_ctx_st EVP_MD_CTX;
               ^
hexpand.c:65:19: error: incomplete definition of type 'struct evp_md_ctx_st'
        ((MD5_CTX *)mdctx->md_data)->B = htonl(strntoul(signature+8, 8, 16));
                    ~~~~~^
/usr/include/openssl/ossl_typ.h:92:16: note: forward declaration of 'struct evp_md_ctx_st'
typedef struct evp_md_ctx_st EVP_MD_CTX;
               ^
hexpand.c:66:19: error: incomplete definition of type 'struct evp_md_ctx_st'
        ((MD5_CTX *)mdctx->md_data)->C = htonl(strntoul(signature+16, 8, 16));
                    ~~~~~^
/usr/include/openssl/ossl_typ.h:92:16: note: forward declaration of 'struct evp_md_ctx_st'
typedef struct evp_md_ctx_st EVP_MD_CTX;
               ^
hexpand.c:67:19: error: incomplete definition of type 'struct evp_md_ctx_st'
        ((MD5_CTX *)mdctx->md_data)->D = htonl(strntoul(signature+24, 8, 16));
                    ~~~~~^
/usr/include/openssl/ossl_typ.h:92:16: note: forward declaration of 'struct evp_md_ctx_st'
typedef struct evp_md_ctx_st EVP_MD_CTX;
               ^
10 errors generated.
make: *** [Makefile:19: hexpand.o] Error 1
```

I read over [the Changes doc for OpenSSL 1.1.0](https://wiki.openssl.org/index.php/OpenSSL_1.1.0_Changes) and puzzled through [the 1.1.1 API docs](https://www.openssl.org/docs/man1.1.1/man3/EVP_MD_CTX_block_size.html) a bit, but I concluded that I don't know enough to provide a pull request to fix this.

--- a/hexpand.c
+++ b/hexpand.c
@@ -19,7 +19,8 @@ unsigned int strntoul(const char* str, int length, int base) {
 }
 
 char* sha1_extend(EVP_MD_CTX *mdctx, char* signature, int length) {
-	int length_modulo = mdctx->digest->block_size;
+	const EVP_MD *md = EVP_MD_CTX_md(mdctx);
+	int length_modulo = EVP_MD_block_size(md);
 	int length_bytes = length_modulo/8;
 	int trunc_length = length%length_modulo;
 	int padding = ((trunc_length) < (length_modulo-length_bytes))
@@ -29,8 +30,8 @@ char* sha1_extend(EVP_MD_CTX *mdctx, char* signature, int length) {
 	memset(data, 'A', length+padding+length_bytes);
 	EVP_DigestUpdate(mdctx, data, length+padding+length_bytes);
 
-	unsigned char* h_data = (unsigned char *)((SHA512_CTX *)mdctx->md_data)->h;
-	int h_data_size = (mdctx->digest->md_size);
+	unsigned char* h_data = (unsigned char *)((SHA512_CTX *)EVP_MD_CTX_md_data(mdctx))->h;
+	int h_data_size = EVP_MD_size(md);
 	int sha_switch = length_modulo/16;
 	int i = 0, j = 0;
 	while (i < h_data_size) {
@@ -44,15 +45,16 @@ char* sha1_extend(EVP_MD_CTX *mdctx, char* signature, int length) {
 	output[0] = '8';
 	output[1] = '0';
 	for (i = 1; i < 2*(padding+length_bytes); i++) output[i] = '0';
-	if (length_modulo == 128) sprintf(output+2*padding, "%032" PRIx32 , htole64(8*length));
-	else sprintf(output+2*padding, "%016" PRIx32 , htole64(8*length));
+	if (length_modulo == 128) sprintf(output+2*padding, "%032lu", htole64(8*length));
+	else sprintf(output+2*padding, "%016lu", htole64(8*length));
 	output[2*(padding+length_bytes)] = 0;
 
 	return output;
 }
 
 char* md5_extend(EVP_MD_CTX *mdctx, char* signature, int length) {
-	int length_modulo = mdctx->digest->block_size;
+	const EVP_MD *md = EVP_MD_CTX_md(mdctx);
+	int length_modulo = EVP_MD_block_size(md);
 	int length_bytes = length_modulo/8;
 	int trunc_length = length&0x3f;
 	int padding = ((trunc_length) < (length_modulo-length_bytes))
@@ -61,10 +63,10 @@ char* md5_extend(EVP_MD_CTX *mdctx, char* signature, int length) {
 	unsigned char data[length+padding+length_bytes];
 	memset(data, 'A', length+padding+length_bytes);
 	EVP_DigestUpdate(mdctx, data, length+padding+length_bytes);
-	((MD5_CTX *)mdctx->md_data)->A = htonl(strntoul(signature, 8, 16));
-	((MD5_CTX *)mdctx->md_data)->B = htonl(strntoul(signature+8, 8, 16));
-	((MD5_CTX *)mdctx->md_data)->C = htonl(strntoul(signature+16, 8, 16));
-	((MD5_CTX *)mdctx->md_data)->D = htonl(strntoul(signature+24, 8, 16));
+	((MD5_CTX *)EVP_MD_CTX_md_data(mdctx))->A = htonl(strntoul(signature, 8, 16));
+	((MD5_CTX *)EVP_MD_CTX_md_data(mdctx))->B = htonl(strntoul(signature+8, 8, 16));
+	((MD5_CTX *)EVP_MD_CTX_md_data(mdctx))->C = htonl(strntoul(signature+16, 8, 16));
+	((MD5_CTX *)EVP_MD_CTX_md_data(mdctx))->D = htonl(strntoul(signature+24, 8, 16));
 
 	int i;
 	char* output = malloc((2*(padding+length_bytes)+1)*sizeof(char));

